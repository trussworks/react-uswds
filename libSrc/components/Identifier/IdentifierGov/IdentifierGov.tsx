import React from 'react'
import classnames from 'classnames'

export interface BaseIdentifierGovProps {
  children?: React.ReactNode
  className?: string
}

export type IdentifierGovProps = BaseIdentifierGovProps &
  JSX.IntrinsicElements['section']

const IdentifierGov = ({
  children,
  className,
  ...sectionProps
}: IdentifierGovProps): React.ReactElement => {
  const classes = classnames(
    'usa-identifier__section usa-identifier__section--usagov',
    className
  )

  return (
    <section data-testid="identifierGov" className={classes} {...sectionProps}>
      <div className="usa-identifier__container">{children}</div>
    </section>
  )
}

export default IdentifierGov
