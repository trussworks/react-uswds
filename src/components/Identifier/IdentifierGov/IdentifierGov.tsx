import React from 'react'
import classnames from 'classnames'

interface IdentifierGovProps {
  children?: React.ReactNode
  className?: string
}

export const IdentifierGov = ({
  children,
  className,
  ...sectionProps
}: IdentifierGovProps &
  JSX.IntrinsicElements['section']): React.ReactElement => {
  const classes = classnames(
    'usa-identifier__section usa-identifier__section--usagov',
    className
  )

  return (
    <section data-testid="identifierGov" className={classes} {...sectionProps}>
      <div className="usa-identifier__container">{children}</div>
    </section>
  )
}

export default IdentifierGov
