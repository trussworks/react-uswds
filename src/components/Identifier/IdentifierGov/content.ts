export default {
  en: {
    ariaLabel: 'U.S. government information and services',
    usaGovDescription: 'Looking for U.S. government information and services?',
    visitGovUrl: 'https://www.usa.gov/',
    visitGovCopy: 'Visit USA.gov',
  },
  es: {
    ariaLabel: 'Información y servicios del Gobierno de EE. UU.',
    usaGovDescription: '¿Necesita información y servicios del Gobierno?',
    visitGovUrl: 'https://www.usa.gov/espanol/',
    visitGovCopy: 'Visite USAGov en Español',
  },
}
