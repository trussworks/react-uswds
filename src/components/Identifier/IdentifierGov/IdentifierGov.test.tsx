import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

import { IdentifierGov } from './IdentifierGov'

describe('IdentifierGov component', () => {
  it('renders without errors', () => {
    const { queryByTestId } = render(<IdentifierGov />)
    expect(queryByTestId('identifierGov')).toBeInTheDocument()
  })

  it('renders section attributes passed in through props', () => {
    const { queryByTestId } = render(
      <IdentifierGov aria-label="custom aria-label value" />
    )

    expect(queryByTestId('identifierGov')).toHaveAttribute(
      'aria-label',
      'custom aria-label value'
    )
  })

  it('renders consistently in Spanish', () => {
    const tree = renderer
      .create(
        <IdentifierGov>
          <div className="usa-identifier__usagov-description">
            ¿Necesita información y servicios del Gobierno?
          </div>
          &nbsp;
          <a href="https://www.usa.gov/espanol/" className="usa-link">
            Visite USAGov en Español
          </a>
        </IdentifierGov>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
