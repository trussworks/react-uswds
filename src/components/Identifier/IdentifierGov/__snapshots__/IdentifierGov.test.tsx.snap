// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdentifierGov component renders consistently in Spanish 1`] = `
<section
  className="usa-identifier__section usa-identifier__section--usagov"
  data-testid="identifierGov"
>
  <div
    className="usa-identifier__container"
  >
    <div
      className="usa-identifier__usagov-description"
    >
      ¿Necesita información y servicios del Gobierno?
    </div>
     
    <a
      className="usa-link"
      href="https://www.usa.gov/espanol/"
    >
      Visite USAGov en Español
    </a>
  </div>
</section>
`;
